### PR TITLE
boolean and port provider improvements

### PR DESCRIPTION
**Pull request by knightorc**
Merge documentation changes
improved why_run? support for boolean and port providers.
Slight change to "persist" changed to attribute rather than a separate action
Handles SELinux gracefully when it is disabled.
Tried to keep as much backwards compatibility as possible.
Added include_recipe to default recipe.
Added rhel-5 vs rhel-6 support.

### CODE REVIEW

**Comment by joerg** [comment]
Whats the status on this? I wrote a simple selinux file provider and wanted to send you a pull request, but if this pull request gets merged I will update it to the new format.

**Comment by BackSlasher** [comment]
I'm still chewing this PR over since it's a lot different than my current philosophy (KISS)
Please submit in the current format, and if I decide to remodel I'll apologize wholeheartedly and see what we can do :)

File: recipes/install.rb [hunk 1]
```diff
@@ -6,13 +6,25 @@
...
     pkgs = [ 'policycoreutils', 'selinux-policy-dev', 'make' ]
   when "rhel"
-    pkgs = [ 'policycoreutils-python', 'selinux-policy', 'make' ]
-  else
-    raise 'Uknown distro, cannot determine required package names'
+    # Handle RHEL 5 vs 6
+    case node['platform_version'].to_i.floor
+      when 5
+        # This pulls in all dependencies
+        pkgs = [ 'policycoreutils', 'selinux-policy', 'make' ]
```
Comment (BackSlasher) [inline recipes/install.rb]: Am I to understand that `policycoreutils-python` is absent in RHEL5?

**Comment by BackSlasher** [comment]
@knightorc what is the motivation for migrating from simple resources to using `shell_out`?
I'm not saying it's bad, just trying to understand the reason

**Comment by knightorc** [comment]
I'll have an answer for your questions this evening.

**Comment by BackSlasher** [comment]
I cherry picked things I could implement myself (and needed now).
The one major thing I avoided is moving out of primitive resources. Please rebase / reply if relevant.
Thanks

**Comment by BackSlasher** [comment]
Closing due to inactivity. Please rebase and create a new one if still relevant

**Pull request closed without merging**